src/sentry/static/sentry/app/views/eventsV2/table/tableView.tsx

```diff
@@ -251,19 +251,40 @@ class TableView extends React.Component<TableViewProps> {
 
       switch (action) {
         case Actions.ADD:
-          // Remove exclusion if it exists.
-          delete query[`!${column.name}`];
-          query[column.name] = [`${value}`];
+          // If the value is null/undefined create a has !has condition.
+          if (value === null || value === undefined) {
+            // Adding a null value is the same as excluding truthy values.
+            if (!query.hasOwnProperty('!has')) {
+              query['!has'] = [];
+            }
+            // Remove inclusion if it exists.
+            if (Array.isArray(query.has) && query.has.length) {
+              query.has = query.has.filter(item => item !== column.name);
+            }
+            query['!has'].push(column.name);
+          } else {
+            // Remove exclusion if it exists.
+            delete query[`!${column.name}`];
+            query[column.name] = [`${value}`];
+          }
           break;
         case Actions.EXCLUDE:
-          // Remove positive if it exists.
-          delete query[column.name];
-          // Negations should stack up.
-          const negation = `!${column.name}`;
-          if (!query.hasOwnProperty(negation)) {
-            query[negation] = [];
+          if (value === null || value === undefined) {
+            // Excluding a null value is the same as including truthy values.
+            if (!query.hasOwnProperty('has')) {
+              query.has = [];
+            }
+            query.has.push(column.name);
```

Comment:
I think we're missing a `remove exclusion if it exists` here

Reply:
Good catch. Originally I thought a user might want to build a query like `!has:thing has:thing` but in hindsight that is a useless query.